tutorial/markdown/aspnet/semantic-kernel/semantic-kernel-tutorial.md

```diff
@@ -0,0 +1,388 @@
+- If omitted (e.g., `IVF,SQ8`), Couchbase auto-selects based on dataset size
+
+**Quantization Options**
+- **SQ** (Scalar Quantization): `SQ4`, `SQ6`, `SQ8` (4, 6, or 8 bits per dimension)
+- **PQ** (Product Quantization): `PQx` (e.g., `PQ32x8`)
+- Higher values = better accuracy, larger index size
+
+**Common Examples**:
+- `IVF,SQ8` - Auto centroids, 8-bit quantization (good default)
+- `IVF1000,SQ6` - 1000 centroids, 6-bit quantization (faster, less accurate)
+- `IVF,PQ32x8` - Auto centroids, product quantization (better accuracy)
+
+For detailed configuration options, see the [Quantization & Centroid Settings](https://docs.couchbase.com/server/current/vector-index/hyperscale-vector-index.html#algo_settings) documentation.
+
+## Running the Demo
+
+### Build and Execute
+```bash
+dotnet build
+dotnet run
+```
+
+### Expected Output
+```
+Couchbase Hyperscale Vector Search Demo
+====================================
+Using OpenAI model: text-embedding-3-small
+Step 1: Ingesting data into Couchbase vector store...
+Data ingestion completed
+
+Step 2: Creating Hyperscale vector index manually...
+Executing Hyperscale index creation query...
+Hyperscale vector index 'hyperscale_glossary_index' already exists.
+
+Step 3: Performing vector search...
+   Found: API
+   Definition: Application Programming Interface. A set of rules and specifications that allow software components to communicate and exchange data.
+   Score: 0.1847
+
+Step 4: Performing filtered vector search...
+   Found (AI category only): RAG
+   Definition: Retrieval Augmented Generation - a term that refers to the process of retrieving additional data to provide as context to an LLM to use when generating a response (completion) to a user's question (prompt).
+   Score: 0.4226
+
+ Demo completed successfully!
+```
+
+## How the Connector Works
+
+The Couchbase Semantic Kernel connector provides a seamless integration between Semantic Kernel's vector store abstractions and Couchbase's vector search capabilities:
+
+### Data Flow
+1. **Initialize** - Create a `CouchbaseVectorStore` instance using a Couchbase scope
+2. **Get Collection** - Use `GetCollection<TKey, TRecord>()` to get a typed collection reference
+3. **Generate Embeddings** - Use Semantic Kernel's `IEmbeddingGenerator` to convert text to vectors
+4. **Upsert Records** - Call `UpsertAsync()` to insert/update records with embeddings
+5. **Create Index** - Set up a vector index using SQL++ for optimal search performance
+6. **Search** - Use `SearchAsync()` with optional `VectorSearchOptions` for filtered searches
+7. **Results** - Receive ranked results with similarity scores (lower = more similar)
+
+### Key Connector Classes & Methods
+
+**Vector Store Classes:**
+- **`CouchbaseVectorStore`** - Main entry point for vector store operations
+- **`CouchbaseQueryCollection`** - Collection class for Hyperscale and Composite indexes (SQL++)
+- **`CouchbaseSearchCollection`** - Collection class for FTS indexes (Search API)
```

Comment:
- Collection class for Search vector indexes (Search, formerly known as Full Text service)